app/src/main/java/com/guruthedev/instagram/ui/fragments/HomeFragment.kt

```diff
@@ -4,39 +4,65 @@ import android.os.Bundle
 import android.view.LayoutInflater
 import android.view.View
 import android.view.ViewGroup
+import android.widget.Toast
 import androidx.fragment.app.Fragment
+import androidx.lifecycle.ViewModelProvider
+import androidx.recyclerview.widget.DividerItemDecoration
 import androidx.recyclerview.widget.LinearLayoutManager
 import com.google.gson.Gson
 import com.guruthedev.instagram.data.InstaStatus
-import com.guruthedev.instagram.data.Post
 import com.guruthedev.instagram.databinding.FragmentHomeBinding
-import com.guruthedev.instagram.ui.adapter.PostAdapter
+import com.guruthedev.instagram.ui.adapter.FeedPostAdapter
 import com.guruthedev.instagram.ui.adapter.StoriesAdapter
+import com.guruthedev.instagram.viewModel.HomeFragmentViewModel
 
 class HomeFragment : Fragment() {
     private lateinit var binding: FragmentHomeBinding
+    private lateinit var adapter: FeedPostAdapter
 
     override fun onCreateView(
         inflater: LayoutInflater, container: ViewGroup?,
         savedInstanceState: Bundle?
     ): View {
         binding = FragmentHomeBinding.inflate(inflater, container, false)
-
         binding.storyRecyclerView.layoutManager =
             LinearLayoutManager(activity, LinearLayoutManager.HORIZONTAL, false)
         binding.storyRecyclerView.setHasFixedSize(true)
-        binding.recyclerViewHome.layoutManager =
-            LinearLayoutManager(activity, LinearLayoutManager.VERTICAL, false)
-        binding.recyclerViewHome.setHasFixedSize(true)
-
         val statusAdapter = StoriesAdapter(getStatus())
         binding.storyRecyclerView.adapter = statusAdapter
-
-        val postAdapter = PostAdapter<Any>(getPostList())
-        binding.recyclerViewHome.adapter = postAdapter
         return binding.root
     }
 
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        initViewModel()
+        initObserver()
+        newInstance()
+    }
+
+    private fun initViewModel() {
+        binding.apply {
+            recyclerViewHome.layoutManager = LinearLayoutManager(activity)
+            val decoration = DividerItemDecoration(activity, DividerItemDecoration.VERTICAL)
+            recyclerViewHome.addItemDecoration(decoration)
+            adapter = FeedPostAdapter()
+            recyclerViewHome.adapter = adapter
+
+        }
+    }
+
+    private fun initObserver() {
+        val viewModel = ViewModelProvider(this).get(HomeFragmentViewModel::class.java)
+        viewModel.getRecycleListObserver().observe(viewLifecycleOwner) { post ->
+            if (post != null) {
```

Comment:
we can use ?.let { } ?: run { }

Reply:
viewModel.getRecycleListObserver().observe(viewLifecycleOwner) { post ->
            if (post != null) {
                feedPostAdapter?.setUpdatedData(post.items)
            } else {
                Toast.makeText(activity, "Error in getting data", Toast.LENGTH_SHORT).show()
            }
        }

we can use ?.let { } ?: run { } how to use this